public/jira/rolledup/work-type/work-type.ts

```diff
@@ -0,0 +1,182 @@
+import {
+  rollupGroupedHierarchy,
+  groupIssuesByHierarchyLevelOrType,
+  zipRollupDataOntoGroupedData,
+  IssueOrRelease,
+  isDerivedRelease,
+} from "../../rollup/rollup";
+import { getStartAndDueData, mergeStartAndDueData } from "../../rollup/dates/dates";
+import {
+  workType as workTypes,
+  WorkType,
+  DerivedWorkStatus,
+} from "../../derived/work-status/work-status";
+import { RollupLevelAndCalculation } from "../../shared/types";
+import { StartData, DueData } from "../../../shared/issue-data/date-data";
+import { DerivedWorkTiming } from "../../derived/work-timing/work-timing";
+
+// TODO:
+
+// this is more like "derived" from "rollup"
+
+// given some "rolled up" dates ....
+
+// Go to each item ... get it's children ... filter by work status type ...
+// add those as children ...
+
+export type DateAndIssueKeys = StartData &
+  DueData & {
+    issueKeys: string[];
+  };
+
+export type WorkTypeRollups = {
+  [key in WorkType]?: Partial<DateAndIssueKeys>;
+};
+
+export type WithChildren = {
+  children: Partial<DateAndIssueKeys>;
+};
+
+export type WithWorkTypeTiming = {
+  workTypeRollups: WorkTypeRollups & WithChildren;
+};
+
+export type WorkTypeChildRollups = {
+  self: WorkTypeRollups;
+  combined: WorkTypeRollups;
+  children: WorkTypeRollups;
+};
+
+function notEmpty<TValue>(value: TValue | null | undefined): value is TValue {
```

Comment:
I found this unintuitive and expected this to need to be 
```
notEmpty<TValue>(value: NonNullable<TValue> | null | undefined): value is NonNullable<TValue> 
```

but I was wrong.  It's inferred correctly that you *meant* "TValue isn't null or undefined here"